### PR TITLE
Set up `ruff`

### PR DESCRIPTION
**Pull request by pamella**
**Description:** Describe in a couple of sentence what this PR adds

Set up `ruff` https://github.com/astral-sh/ruff

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

N/A

**Merge deadline:** List merge deadline (if any)

N/A

**Installation instructions:** List any non-trivial installation 
instructions.

N/A

**Testing instructions:**

- CI is passing

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: .isort.cfg
Comment (pamella) [inline .isort.cfg:1]: Changes made to `isort` so it can work with `ruff`.

File: .ruff.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    ".pyenv",
+    ".pytest_cache",
+    "*/__pycache__/*",
+    "*/migrations/*",
+    "docs",
+]
+
+line-length = 100
+
+target-version = "py38"
```
Comment (pamella) [inline .ruff.toml:13]: Older version supported.

File: manage.py
Comment (pamella) [inline manage.py:1]: Drive-by change:  small refactor to address Django 5.0 changes.

File: .isort.cfg
Comment (fjsj) [reply .isort.cfg:1]: I think you can drop isort as well or put its settings inside ruff:
- https://news.ycombinator.com/item?id=34063467
- https://docs.astral.sh/ruff/settings/#lintisort

**Review by fjsj — APPROVED** [review]
LGTM, but we can drop isort as well.

File: .isort.cfg
Comment (pamella) [reply .isort.cfg:1]: Nice! I updated the code to use those settings

**Review by fjsj — APPROVED** [review]
